### ISSUE TITLE
Suggestion: Warn when using `this` in function without binding it first

### ISSUE THREAD

**Issue by vjeux** [eslint/eslint#861]
A recurring issue with React newcomers is the following

``` javascript
var Component = React.createClass({
  onClick: function() {
    // some code
  },
  render: function() {
    return React.DOM.div(null,
      [1, 2, 3].map(function(i) {
        return React.DOM.button({onClick: this.onClick}, i);
      });
   });
});
```

Unfortunately, if you don't do `}.bind(this)` in the map function, then it's going to look at `window.onClick` and returns undefined, which React treats it as non-existent.

Would it be possible to warn in this case since it's never a good idea to use `this` without the function being bound to something.
